frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js

```diff
@@ -260,6 +266,11 @@ function getDimensionsAndGroupsForWaterfallChart(props, originalDatas, warn) {
     datas[2][k][1] = positives[k];
   }
 
+  // The last one is the total bar, treat it as either a positive or negative bar
+  datas[0][last][1] = 0;
+  datas[1][last][1] = totalValue < 0 ? totalValue : 0;
+  datas[2][last][1] = totalValue > 0 ? totalValue : 0;
+
```

Comment:
Since `last` is in range for the above loops, didn't this already happen?

Reply:
It didn't happen, since the Total bar needs a special case (it's "anchored" at 0, not "floated" like the other positives or negatives).
OTOH, this block of code is really messy. My follow-up is to refactor it to the utils file, add some unit tests, and hopefully the mechanics will be easier understood from the tests themselves.